test/dlc_tests/release_candidate_integration/test_sm_profiler.py

```diff
@@ -0,0 +1,126 @@
+    if processor not in ("cpu", "gpu"):
+        pytest.skip(f"Processor {processor} not supported. Skipping test.")
+
+    ctx = Context()
+
+    profiler_tests_dir = os.path.join(
+        os.getenv("CODEBUILD_SRC_DIR"), get_container_name("smprof", tensorflow_training), "smprofiler_tests"
+    )
+    ctx.run(f"mkdir -p {profiler_tests_dir}")
+
+    # Download sagemaker-tests zip
+    sm_tests_zip = "sagemaker-tests.zip"
+    ctx.run(f"aws s3 cp s3://smprofiler-test-artifacts/{sm_tests_zip} {profiler_tests_dir}/{sm_tests_zip}", hide=True)
+    ctx.run(f"cd {profiler_tests_dir} && unzip {sm_tests_zip}", hide=True)
+
+    # Install tf datasets
+    ctx.run(
+        f"echo 'tensorflow-datasets==4.0.1' >> {profiler_tests_dir}/sagemaker-tests/tests/scripts/tf_scripts/requirements.txt",
+        hide=True,
+    )
+
+    run_sm_profiler_tests(tensorflow_training, profiler_tests_dir, "test_profiler_tensorflow.py", processor)
+
+
+def run_sm_profiler_tests(image, profiler_tests_dir, test_file, processor):
+    ctx = Context()
+    smdebug_version = ctx.run(
+        f"docker run {image} python -c 'import smdebug; print(smdebug.__version__)'", hide=True
+    ).stdout.strip()
+
+    if Version(smdebug_version) < Version("1"):
+        pytest.skip(f"smdebug version {smdebug_version} is less than 1, so smprofiler not expected to be present")
+
+    # Install smprofile requirements from GitHub
+    ctx.run(
+        f"pip install -r https://raw.githubusercontent.com/awslabs/sagemaker-debugger/master/config/profiler/requirements.txt",
+        hide=True,
+        warn=True,
+    )
+
+    # Collect env variables for tests
+    framework, version = get_framework_and_version_from_tag(image)
+    spec_file = f"buildspec_profiler_sagemaker_{framework}_{version.replace('.', '_')}_integration_tests.yml"
+
+    # Get buildspec file from GitHub
+    # Note: SMDebug seems to update these in master, not necessarily in feature branches, hence using master branch
+    ctx.run(
+        f"wget https://raw.githubusercontent.com/awslabs/sagemaker-debugger/master/config/profiler/{spec_file}",
+        hide=True,
+    )
+    with open(spec_file, "r") as sf:
+        yml_envs = yaml.safe_load(sf)
+        spec_file_envs = yml_envs.get("env", {}).get("variables")
+
+    # Command to set all necessary environment variables
+    export_cmd = " && ".join(f"export {key}={val}" for key, val in spec_file_envs.items())
+    export_cmd = f"{export_cmd} && export ENV_{processor.upper()}_TRAIN_IMAGE={image}"
+
+    test_results_outfile = os.path.join(os.getcwd(), f"{get_container_name('smprof', image)}.txt")
+    with ctx.prefix(f"cd {profiler_tests_dir}"):
+        with ctx.prefix(f"cd sagemaker-tests/tests && {export_cmd}"):
+            test_output = ctx.run(
+                f"pip install smdebug && pytest --json-report --json-report-file={test_results_outfile} -n=auto -v -s -W=ignore {test_file}::test_{processor}_jobs",
```

Comment:
Are we installing `smdebug` on the host machine? why is that required?

Reply:
This may no longer be required, but in the original script, smdebug is a test requirement, but not specified in their test requirements explicitly. I can try without it

Reply:
do we need a rerun here?